contents/docs/self-host/deploy/snippets/upgrading.mdx

```diff
@@ -15,19 +15,51 @@ helm search repo posthog
 ```
 Compare the numbers for the chart version (in the format `posthog-{major}.{minor}.{patch}` - for example, `posthog-3.15.1`) when running the commands above. If the upgrade is for a major version, check the [upgrade notes](/docs/self-host/deploy/upgrade-notes) before moving forward.
 
-3. Run the upgrade
+3. (optional) Grab the up-to-date values.yaml file and save to values.yaml
+
+```shell
+helm get values posthog -n posthog
+```
+
+4. Run the upgrade
 
 <CommandHelmUpgradeSnippet />
 
 Check the Helm [documentation](https://helm.sh/docs/helm/helm_upgrade/) for more info about the `helm upgrade` command.
 
 <details>
 <summary>Troubleshooting</summary>
-If you see this error
+If you see this error:
 
 ```
 Error: UPGRADE FAILED: release posthog failed, and has been rolled back due to atomic being set: post-upgrade hooks failed: warning: Hook post-upgrade posthog/templates/migrate.job.yaml failed: jobs.batch "posthog-migrate" already exists
 ```
 
-it happens because the migrate job was left around from a previous upgrade attempt, we just need to kill that job (`kubectl -n posthog delete job $(kubectl -n posthog get jobs --no-headers -o custom-columns=NAME:.metadata.name)`) before running the upgrade again.
+It happens because the migrate job was left around from a previous upgrade attempt, we just need to kill that job (`kubectl -n posthog delete job $(kubectl -n posthog get jobs --no-headers -o custom-columns=NAME:.metadata.name)`) before running the upgrade again.
+  
+---
+
+If you see this error:
+
+```
+Error: UPGRADE FAILED: an error occurred while rolling back the release. original upgrade error: cannot patch "posthog-posthog-kafka" with kind StatefulSet: StatefulSet.apps "posthog-posthog-kafka" is invalid
```

Comment:
We should likely make this more generic about any sts & link to runbook which explains how to size up/down.